### PR TITLE
Update in order to use a reactor pattern

### PR DESCRIPTION
**Pull request by antechrestos**
This change aims to use a reactor pattern in order to do bindings in a parallel way.

### CODE REVIEW

File: src/main/java/org/cloudfoundry/autosleep/worker/remote/CloudFoundryApi.java [hunk 1]
```diff
@@ -101,46 +112,39 @@ public void onSubscribe(Subscription subscription) {
...
+        final Subscriber<U> subscriber
+                = new BaseSubscriber<>(latch, errorEncountered::set, null);
+
+        objectsToBind.forEach(objectToBind -> caller.apply(objectToBind).subscribe(subscriber));
+        waitForResult(latch, errorEncountered, null);
     }
 
     @Override
-    public void bindServiceInstance(List<ApplicationIdentity> applications, String serviceInstanceId) throws
+    public void bindApplications(String serviceInstanceId, List<ApplicationIdentity> applications) throws
```
Comment (arnaudruffin) [inline src/main/java/org/cloudfoundry/autosleep/worker/remote/CloudFoundryApi.java:128]: I think we should reconsider the method name. **bindApplications** is ambiguous. We either bind a _service_ or a _route_, but always to an _application_

Comment (antechrestos) [inline src/main/java/org/cloudfoundry/autosleep/worker/remote/CloudFoundryApi.java:128]: I do not agree: if you look at the code, the two resources that are bound are either a service dans an application, or a service and a route. Truely, he route belongs to an application, but it is an other entity.

Comment (arnaudruffin) [inline src/main/java/org/cloudfoundry/autosleep/worker/remote/CloudFoundryApi.java:128]: You're right, my mistake

**Comment by arnaudruffin** [comment]
Really nice work, thanks!

**Pull request merged**